tests/conftest.py

```diff
@@ -238,3 +239,38 @@ def unprivileged_client(
 
         else:
             yield admin_client
+
+
+@pytest.fixture(scope="session")
+def dsc_resource(admin_client: DynamicClient, name: str = "default-dsc"):
+    for dsc in DataScienceCluster.get(dyn_client=admin_client, name=name):
+        yield dsc
+
+
+def get_dsc_component_state(dsc_resource: DataScienceCluster, component_name: str) -> str:
+    return dsc_resource.instance.spec.components[component_name].managementState
+
+
+@pytest.fixture(scope="module")
+def set_dsc_component_state(
+    dsc_resource: DataScienceCluster,
+    request: FixtureRequest,
+) -> Generator[DataScienceCluster, Any, Any]:
+    component_name = getattr(request.module, "component_name")
+    desired_state = getattr(request.module, "desired_state")
+    if get_dsc_component_state(dsc_resource=dsc_resource, component_name=component_name) != desired_state:
+        with ResourceEditor(
```

Comment:
note that  `ResourceEditor` does the check

```
LOGGER.warning(f"ResourceEdit: no diff found in patch for {resource.name} -- skipping")
```

Reply:
Mh, but if you didn't know you wouldn't be able to tell from looking at this code. Honestly I'd rather keep it explicit here.